#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_compile-with-g++-4.3.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## DP: Fixes for compiling octave2.1 with g++-4.3

@DPATCH@

--- octave2.1-2.1.73.orig/liboctave/data-conv.cc
+++ octave2.1-2.1.73/liboctave/data-conv.cc
@@ -26,6 +26,7 @@
 #endif
 
 #include <cctype>
+#include <cstdlib>
 
 #include <iostream>
 
--- octave2.1-2.1.73.orig/liboctave/lo-utils.cc
+++ octave2.1-2.1.73/liboctave/lo-utils.cc
@@ -29,6 +29,7 @@
 #include <climits>
 #include <cstdlib>
 #include <cstdio>
+#include <cstring>
 
 #include <string>
 
--- octave2.1-2.1.73.orig/liboctave/oct-env.cc
+++ octave2.1-2.1.73/liboctave/oct-env.cc
@@ -43,6 +43,7 @@
 
 #include <cctype>
 #include <cstdlib>
+#include <cstring>
 
 #include <string>
 
--- octave2.1-2.1.73.orig/src/unwind-prot.cc
+++ octave2.1-2.1.73/src/unwind-prot.cc
@@ -30,6 +30,7 @@
 #endif
 
 #include <cstddef>
+#include <cstring>
 
 #include "CMatrix.h"
 
--- octave2.1-2.1.73.orig/src/pr-output.cc
+++ octave2.1-2.1.73/src/pr-output.cc
@@ -32,6 +32,7 @@
 
 #include <iomanip>
 #include <iostream>
+#include <sstream>
 #include <string>
 
 #include "Array-util.h"
@@ -1602,7 +1603,7 @@
 }
 
 template <>
-static inline void
+/* static */ inline void
 pr_plus_format (std::ostream& os, const Complex& c)
 {
   double rp = c.real ();
--- octave2.1-2.1.73.orig/src/DLD-FUNCTIONS/sort.cc
+++ octave2.1-2.1.73/src/DLD-FUNCTIONS/sort.cc
@@ -26,6 +26,8 @@
 #include <config.h>
 #endif
 
+#include <vector>
+
 #include "lo-mappers.h"
 #include "quit.h"
 
@@ -294,7 +296,7 @@
 template class octave_sort<vec_index<unsigned EIGHT_BYTE_INT> *>;
 
 template <>
-static octave_value
+/* static */ octave_value
 mx_sort (ArrayN<double> &m, int dim, sortmode mode)
 {
   octave_value retval;
@@ -431,7 +433,7 @@
 }
 
 template <>
-static octave_value_list
+/* static */ octave_value_list
 mx_sort_indexed (ArrayN<double> &m, int dim, sortmode mode)
 {
   octave_value_list retval;
--- octave2.1-2.1.73.orig/liboctave/oct-sort.cc
+++ octave2.1-2.1.73/liboctave/oct-sort.cc
@@ -84,6 +84,8 @@
 #include <config.h>
 #endif
 
+#include <cstdlib>
+
 #include "lo-mappers.h"
 #include "quit.h"
 #include "oct-sort.h"
--- octave2.1-2.1.73.orig/liboctave/chMatrix.cc
+++ octave2.1-2.1.73/liboctave/chMatrix.cc
@@ -30,6 +30,8 @@
 #include <config.h>
 #endif
 
+#include <cstring>
+
 #include <iostream>
 #include <string>
 
